src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.diff.sql

```diff
@@ -133,113 +168,143 @@ CREATE UNIQUE CLUSTERED INDEX IXC_ExtendedQueryTagDateTime ON dbo.ExtendedQueryT
           the value on the instance with the highest watermark wins.
     Note: The primary key is designed on the assumption that tags only occur once in an instance.
 **************************************************************/
-CREATE TABLE dbo.ExtendedQueryTagPersonName (
-    TagKey                  INT                  NOT NULL, --FK
-    TagValue                NVARCHAR(200)        COLLATE SQL_Latin1_General_CP1_CI_AI NOT NULL,
-    StudyKey                BIGINT               NOT NULL, --FK
-    SeriesKey               BIGINT               NULL,     --FK
-    InstanceKey             BIGINT               NULL,     --FK
-    Watermark               BIGINT               NOT NULL,
-    WatermarkAndTagKey      AS CONCAT(TagKey, '.', Watermark), --PK
-    TagValueWords           AS REPLACE(REPLACE(TagValue, '^', ' '), '=', ' ') PERSISTED,
-) WITH (DATA_COMPRESSION = PAGE)
-
-CREATE UNIQUE CLUSTERED INDEX IXC_ExtendedQueryTagPersonName ON dbo.ExtendedQueryTagPersonName
-(
-    TagKey,
-    TagValue,
-    StudyKey,
-    SeriesKey,
-    InstanceKey
-)
-
-CREATE UNIQUE NONCLUSTERED INDEX IXC_ExtendedQueryTagPersonName_WatermarkAndTagKey ON dbo.ExtendedQueryTagPersonName
-(
-    WatermarkAndTagKey
-)
-
-CREATE FULLTEXT INDEX ON ExtendedQueryTagPersonName(TagValueWords LANGUAGE 1033)
-KEY INDEX IXC_ExtendedQueryTagPersonName_WatermarkAndTagKey
-WITH STOPLIST = OFF;
+IF NOT EXISTS (
```

Comment:
Wrapping the if on table and index is not completely correct.

Table may exist and the index which is a different object maybe missing.

Reply:
Good catch, thanks! Fixed.